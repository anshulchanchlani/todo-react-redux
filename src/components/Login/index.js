import React, { Component } from 'react';
import { Container, Row, Col } from 'react-grid-system';
import './index.scss'
class Login extends Component {
    constructor(props) {
        super(props);
        this.state = { username: '', password: '',error:false };
        this.handleOnChange = this.handleOnChange.bind(this);
        this.handleLogin = this.handleLogin.bind(this);
    }
    handleOnChange(e) {
        this.setState({error:false})
        if (e.target.value && e.target.value !== '') {
            this.setState({ [e.target.name]: e.target.value });

        }
    }
    handleLogin(e) {
        e.preventDefault();
        if (this.state.username !== '' && this.state.password !== '') {
           
             this.props.loginHandler(this.state.username,this.state.password)
        }else{
            this.setState({error:true})
        }

    }
    render() {

        return (
            <Container className="loginContainer">
                <form onSubmit={this.handleLogin}>
                <Row>
                    <Col xs={12} md={8} lg={8} sm={12}>
                        <input type="text" name="username" placeholder="Enter your username" onChange={this.handleOnChange} />
                    </Col>
                </Row>
                <Row>

                    <Col xs={12} md={8} lg={8} sm={12}>
                        <input type="password" name="password" placeholder="Enter your password" onChange={this.handleOnChange} />
                    </Col>
                </Row>
                <Row>
                    <Col xs={12} md={8} lg={8} sm={12}>
                        <button  onClick={this.handleLogin}>Login</button>
                    </Col>
                </Row>
                {this.state.error?<Row>
                    <Col xs={12} md={8} lg={8} sm={12}>
                        <span className="error-msg">Please check your credentials.</span>
                    </Col>
                    
                </Row>:null}
                </form>
            </Container>
        );
    }
}

export default Login;