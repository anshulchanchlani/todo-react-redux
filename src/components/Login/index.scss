.loginContainer{
    
    width:80%;
    div{
        display:flex;
        justify-content: space-around !important;
        
    }
    input{
        text-align: center;
        width:100%;
        height:40px;
        margin:10px;
    }
    button{
        text-align: center;
        width:100%;
        height:40px;
        margin:10px;
        cursor:pointer;
    }
    .error-msg{
        width:100%;
        text-align:center;
        color:red;
    }
}