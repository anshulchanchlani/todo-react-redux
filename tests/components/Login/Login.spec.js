import React from 'react';
import Login from '../../../src/components/Login';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import {shallow} from 'enzyme';


describe('Login tests',()=>{
    const wrapper = shallow(<Login/>); 
    test('Login renders',()=>{
      expect(toJson(wrapper)).toMatchSnapshot();
    })

    test('Login has a login form',()=>{
        expect(wrapper.find('form').length).toEqual(1)
        expect(wrapper.find('input').length).toEqual(2)
        expect(wrapper.find('button').length).toEqual(1)
    })
})
