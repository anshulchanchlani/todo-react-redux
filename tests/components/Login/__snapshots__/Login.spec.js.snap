// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login tests Login renders 1`] = `
<Container
  className="loginContainer"
  component="div"
  fluid={false}
  lg={false}
  md={false}
  sm={false}
  style={Object {}}
  xl={false}
  xs={false}
>
  <form
    name="Login Form"
    onSubmit={[Function]}
  >
    <Row
      align="normal"
      component="div"
      debug={false}
      justify="start"
      nogutter={false}
      style={Object {}}
    >
      <Col
        component="div"
        debug={false}
        lg={8}
        md={8}
        offset={Object {}}
        pull={Object {}}
        push={Object {}}
        sm={12}
        style={Object {}}
        xl={null}
        xs={12}
      >
        <input
          aria-label="username"
          name="username"
          onChange={[Function]}
          placeholder="Enter your username"
          type="text"
        />
      </Col>
    </Row>
    <Row
      align="normal"
      component="div"
      debug={false}
      justify="start"
      nogutter={false}
      style={Object {}}
    >
      <Col
        component="div"
        debug={false}
        lg={8}
        md={8}
        offset={Object {}}
        pull={Object {}}
        push={Object {}}
        sm={12}
        style={Object {}}
        xl={null}
        xs={12}
      >
        <input
          aria-label="password"
          name="password"
          onChange={[Function]}
          placeholder="Enter your password"
          type="password"
        />
      </Col>
    </Row>
    <Row
      align="normal"
      component="div"
      debug={false}
      justify="start"
      nogutter={false}
      style={Object {}}
    >
      <Col
        component="div"
        debug={false}
        lg={8}
        md={8}
        offset={Object {}}
        pull={Object {}}
        push={Object {}}
        sm={12}
        style={Object {}}
        xl={null}
        xs={12}
      >
        <button
          aria-label="Login"
          name="Login"
          onClick={[Function]}
        >
          Login
        </button>
      </Col>
    </Row>
  </form>
</Container>
`;
